ls1intum/Artemis
# `Exam mode`: Improve the user interface of the exam timeline

### Checklist
#### General


#### Client

### Motivation and Context


The UI of the exam timeline can be improved.


### Description

- addded comments requested by @pal03377 
- do no longer hide button if student exam is not submitted but disable it and show a tooltip why it's disabled
- show the commit-hash that belongs to the submissions directly next to the view button for the git-diff.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exam with at least one programming exercise with one submitted and one not submitted exam.

1. Verify that the exam timeline button on the student exam details page is disabled for the not submitted exam and a tooltip is shown that explains when the button will be enabled.
2. Verify that the exam timeline button on the student exam details page is enabled for the submitted student exam and the view for a programming exercise in the exam timeline contains the commit-hash next to the view button.

#### Exam Mode Testing
Covered with the testing steps above.

### Review Progress






#### Performance Review

#### Code Review
#### Manual Tests

### Test Coverage






Only UI changes in html files.

### Screenshots



https://github.com/ls1intum/Artemis/assets/84102468/52cd9be8-b413-404b-9f45-e25381b8e00c
